widget/playground/src/components/TokensPanel/TokensPanel.tsx

```diff
@@ -36,10 +36,96 @@ export function TokensPanel(props: PropTypes) {
 
   const [list, setList] = useState(listProps);
 
-  const [excludeList, setExcludeList] = useState(configTokens || {});
+  const [supportedTokenList, setSupportedTokenList] = useState(
+    configTokens || {}
+  );
   const [showSelectedTokens, setShowSelectedTokens] = useState(false);
 
+  const getTokens = (
+    selected: boolean,
+    allTokensInBlockchain: Asset[],
+    token?: TokenType
+  ) => {
+    if (!token) {
+      return selected ? allTokensInBlockchain : [];
+    }
+    const { blockchain, symbol, address } = token;
+    if (!selected) {
+      return [
+        ...supportedTokenList[blockchain].tokens.filter(
+          (t) => !tokensAreEqual(t, token)
+        ),
+      ];
+    }
+    return [
+      ...supportedTokenList[blockchain].tokens,
+      { symbol, address, blockchain },
+    ];
+  };
+
+  const makeSupportedTokenList = (
```

Comment:
If you add a comment for this function, it will become more readable.

Reply:
fixed